demo/chaincode/fpc/Makefile

```diff
@@ -20,3 +20,7 @@ build: $(BUILD_DIR)
 
 clean:
 	-rm -rf $(BUILD_DIR)
+
+docker-build: clean
```

Comment:
Oh, I don't need to run as root to run clean. Looking at the permissions on the directory they seem to have all the permissions the other files do. `drwxr-xr-x` vs `-rw-r--r--`. Is it better to just remove `clean` from this target? I am guessing this could be a different between mac vs linux? I do have root privileges but I don't run these commands with sudo.

Reply:
it depends on how your docker is setup. Presumably, on your mac docker will map docker-internal root to your personal id (rather than external root by default for linux docker)?  You could run, using the -u option, docker as the user calling it instead of running as root. Fabric does that in some script, which unfortunately caused me then problem with GOCACHE trying to write files in / which resulted in permission denied (because that is of course owned by root). defining GOCACHE env variable would solve it (as might setting HOME?) but this shouldn't be an issue for cc-builder as it's all C?  (unfortunately, i can't remember which fabric script it was which did the ID-mapping ..)

Reply:
In any case, just removing the clean dependency doesn't really solve the problem as you still will have root-owned `_build` directory after it got build using docker, so any other clean will fail there ...  So probably doing the id-mapping in docker would be best (another maybe simpler hack could be to just in cc-builder change permissions of `_build/*` to world-writable (as much as this make cringe to propose this as a security person :-))

Reply:
I also have this problem. Using `-u` makes sense to me.

Reply:
Let me play around with the id-mapping. I am with you, I rather not change the permissions. I also want people to see how we do it with this demo and possibly replicate this process for other chaincodes they may want to build so better to have good practices.